google/cloud/iam_bindings.h

```diff
@@ -0,0 +1,151 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_IAM_BINDINGS_H_
+#define GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_IAM_BINDINGS_H_
+
+#include "google/cloud/iam_binding.h"
+#include "google/cloud/version.h"
+#include <map>
+#include <set>
+#include <vector>
+
+namespace google {
+namespace cloud {
+
+/**
+ * Represents a container for providing users with a handful of operation to
+ * users which they can use to add and remove members to Binding which is used
+ * for defining IAM Policy for Cloud Platform Resources.
+ *
+ * For more information about a Binding please refer to:
+ * https://cloud.google.com/resource-manager/reference/rest/Shared.Types/Binding
+ */
+class IamBindings {
+ public:
+  IamBindings(std::vector<IamBinding> bindings) {
+    for (auto& it : bindings) {
+      bindings_.insert({std::move(it.role()), std::move(it.members())});
+    }
+  }
+
+  IamBindings(std::string const& role, std::set<std::string> const& members) {
+    bindings_.insert({std::move(role), std::move(members)});
+  }
+
+  using iterator = std::map<std::string, std::set<std::string>>::iterator;
+  /**
+   * Returns an iterator referring to the first element in IamBindings
+   * container.
+   */
+  iterator begin() {
```

Comment:
That is a mutable iterator, did you intend that? If so, we need an overload for `const_iterator begin() const`.